src/main/java/net/fabricmc/loader/EntrypointStorage.java

```diff
@@ -145,20 +157,39 @@ boolean hasEntrypoints(String key) {
 					results.add(result);
 				}
 			} catch (Throwable t) {
-				errors.add(t);
+				errors.put(entry, t);
 			}
 		}
 
 		if (!errors.isEmpty()) {
 			EntrypointException e = new EntrypointException("Could not look up entries for entrypoint " + key + "!");
 
-			for (Throwable suppressed : errors) {
-				e.addSuppressed(suppressed);
+			for (Map.Entry<Entry, Throwable> suppressed : errors.entrySet()) {
+				e.addSuppressed(new EntrypointException("Caused by mod: '" + suppressed.getKey().getModContainer().getMetadata().getId() + "' ", suppressed.getValue()));
```

Comment:
You should integrate the changes in #180 and add the other exceptions as suppressed by the first one.